### PR TITLE
Funding support in frontmatter (and other small JATS features)

### PR DESCRIPTION
**Pull request by fwkoch**
In Frontmatter:
- Add funding field
- Split out authors and contributors to support referencing contributors by id in funding object

In JATS
- support funding
- support parsed names
- support keywords
- support contributors (in addition to authors)
- support full affiliation objects

### CODE REVIEW

File: packages/myst-frontmatter/src/frontmatter/validators.ts [hunk 1]
```diff
@@ -513,15 +515,19 @@ export function validateName(input: any, opts: ValidationOptions) {
 }
 
 /**
- * Validate Author object against the schema
+ * Validate Conributor object against the schema
```
Comment (fwkoch) [inline packages/myst-frontmatter/src/frontmatter/validators.ts:518]:
```suggestion
 * Validate Contributor object against the schema
```

File: packages/myst-frontmatter/src/frontmatter/validators.ts [hunk 2]
```diff
@@ -389,9 +394,6 @@ export function validateAffiliation(input: any, opts: ValidationOptions) {
   if (defined(value.name)) {
     output.name = validateString(value.name, incrementOptions('name', opts));
   }
-  if (defined(value.institution)) {
-    output.institution = validateString(value.institution, incrementOptions('institution', opts));
-  }
   if (defined(value.department)) {
     output.department = validateString(value.department, incrementOptions('department', opts));
```
Comment (stevejpurves) [inline packages/myst-frontmatter/src/frontmatter/validators.ts:398]: does this have an impact of fields available to exports, e.g. web, latex templates...?

File: packages/myst-frontmatter/src/frontmatter/validators.ts [hunk 3]
```diff
@@ -533,7 +539,7 @@ export function validateAuthor(input: any, stash: ReferenceStash, opts: Validati
     output.nameParsed = validateName(value.name, incrementOptions('name', opts));
     output.name = output.nameParsed?.literal;
   } else {
-    validationWarning('author should include name', opts);
+    validationWarning('contributor should include name', opts);
```
Comment (stevejpurves) [inline packages/myst-frontmatter/src/frontmatter/validators.ts:542]: I wonder if this is less meaningful for the author now. For example if there is an error in frontmatter in the `authors` field, will the message here be talking about contributors? That could be misleading

Comment (stevejpurves) [reply packages/myst-frontmatter/src/frontmatter/validators.ts:542]: can a user now list `authors` and `contributors` in their frontmatter?

File: packages/myst-frontmatter/src/frontmatter/validators.ts [hunk 4]
```diff
@@ -1068,7 +1078,31 @@ export function validateSiteFrontmatterKeys(value: Record<string, any>, opts: Va
...
+  if (defined(value.funding)) {
+    const funding = Array.isArray(value.funding) ? value.funding : [value.funding];
+    output.funding = validateList(funding, incrementOptions('funding', opts), (fund, index) => {
+      return validateFunding(fund, stash, incrementOptions(`funding.${index}`, opts));
+    });
+  }
+  const stashAuthors = stash.contributors?.filter(
+    (contrib) => stash.authorIds?.includes(contrib.id),
+  );
+  const stashContributors = stash.contributors?.filter(
```
Comment (stevejpurves) [inline packages/myst-frontmatter/src/frontmatter/validators.ts:1090]: `NonAuthorContributors`?

**Comment by fwkoch** [comment]
See #596

**Pull request closed without merging**